### PR TITLE
fix(checkbox): address typo

### PR DESCRIPTION
**Pull request by jenndiaz**
## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [x] No visual changes to the docs site 
- [x] No visual changes storybook 
- [x] should be able to use `--mod-checkbox-checkmark-color` to change the color of the partial checkbox (the dash)

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [x] VRTs have been run and looked at.
- [x] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [ ] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [ ] I have updated relevant storybook stories and templates.
- [ ] I have tested these changes in Windows High Contrast mode.
- [ ] If my change impacts **other components**, I have tested to make sure they don't break.
- [ ] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/checkbox/index.css [hunk 1]
```diff
@@ -533,7 +533,7 @@ governing permissions and limitations under the License.
 
   .spectrum-Checkbox-checkmark,
   .spectrum-Checkbox-partialCheckmark {
-    color: var(--highcontrast-checkbox-background-color-default, var(--mode-checkbox-checkmark-color, var(--spectrum-checkbox-checkmark-color)));
+    color: var(--highcontrast-checkbox-background-color-default, var(--mod-checkbox-checkmark-color, var(--spectrum-checkbox-checkmark-color)));
```
Comment (jenndiaz) [inline components/checkbox/index.css:536]: @pfulton are we ok not having the incorrect spelling retained as a fallback in this case?

**Review by mdt2 — APPROVED** [review]
✨

**Review by castastrophe — APPROVED** [review]

**Comment by castastrophe** [comment]
@jenndiaz Shouldn't the mods file have had an update?

File: components/checkbox/index.css (hunk 1 above)
Comment (pfulton) [reply components/checkbox/index.css:536]: Yeah, I think that's fine. After we merge and release this, I can let the bug reporter know about this change.

**Review by pfulton — APPROVED** [review]
This looks good. Can you double-check to see if the mods file is getting this change after you regenerate it?

**Comment by jenndiaz** [comment]
> @jenndiaz Shouldn't the mods file have had an update?
@pfulton and @castastrophe 

It does not need updating, the [mods file already](https://github.com/adobe/spectrum-css/blob/main/components/checkbox/metadata/mods.md) had the correct spelling already since  `--mod-checkbox-checkmark-color` is used elsewhere in the `index.css` file

**Pull request merged**